rosbag2_transport/src/rosbag2_transport/formatter.cpp

```diff
@@ -90,7 +90,7 @@ std::string Formatter::format_time_point(
          " (" + formatted_duration["time_in_sec"] + ")";
 }
 
-std::string Formatter::format_file_size(size_t file_size)
+std::string Formatter::format_file_size(uint64_t file_size)
```

Comment:
This seems better to me.

However, we have a further problem higher in the stack.  The only caller of this method in real code is [formatter.cpp](https://github.com/ros2/rosbag2/blob/master/rosbag2_transport/src/rosbag2_transport/formatter.cpp#L42), where it passes a `BagMetadata.bag_size` into the function.  [BagMetadata](https://github.com/ros2/rosbag2/blob/master/rosbag2_storage/include/rosbag2_storage/bag_metadata.hpp#L34) defines the bag_size as a `size_t`, which is also incorrect.  The real issue is that we are now breaking on-disk serialized code, but using a `size_t` in here was always a bug; it will be a different size on different platforms.  My suggestion would be to bump the version to "2", change the bag_size to a `uint64_t`, and also change the message_count to a `uint64_t`; then it will be more cross-platform.  However, that is a big change to the whole ecosystem, so I'd like to get a further opinion from @Karsten1987 before moving ahead with that.

Reply:
Yes, I was looking at [metadata_io.cpp](https://github.com/ros2/rosbag2/blob/master/rosbag2_storage/src/rosbag2_storage/metadata_io.cpp#L111) which is similarly serializing `metadata.yaml` test resource files incorrectly into `size_t` instead of `uint64_t`. That is causing test failures in [rosbag2_tests](https://ci.ros2.org/job/ci_linux-armhf/17/testReport/junit/rosbag2_tests/).

I have a similar fix for it that uses the correct variable type, but I'd like for us to decide on an approach before I open that PR.